azure/scope/identity.go

```diff
@@ -299,28 +299,51 @@ func createAzureIdentityWithBindings(ctx context.Context, azureIdentity *infrav1
 		return errors.Errorf("failed to create copied AzureIdentity %s in %s: %v", copiedIdentity.Name, system.GetManagerNamespace(), err)
 	}
 
-	azureIdentityBinding := &aadpodv1.AzureIdentityBinding{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "AzureIdentityBinding",
-			APIVersion: "aadpodidentity.k8s.io/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%s-binding", copiedIdentity.Name),
-			Namespace: copiedIdentity.Namespace,
-			Labels: map[string]string{
-				clusterv1.ClusterNameLabel:              clusterMeta.Name,
-				infrav1.ClusterLabelNamespace:           clusterMeta.Namespace,
-				clusterctl.ClusterctlMoveHierarchyLabel: "true",
+	bindings := []*aadpodv1.AzureIdentityBinding{
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "AzureIdentityBinding",
+				APIVersion: "aadpodidentity.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-binding", copiedIdentity.Name),
+				Namespace: copiedIdentity.Namespace,
+				Labels: map[string]string{
+					clusterv1.ClusterNameLabel:              clusterMeta.Name,
+					infrav1.ClusterLabelNamespace:           clusterMeta.Namespace,
+					clusterctl.ClusterctlMoveHierarchyLabel: "true",
+				},
+			},
+			Spec: aadpodv1.AzureIdentityBindingSpec{
+				AzureIdentity: copiedIdentity.Name,
+				Selector:      infrav1.AzureIdentityBindingSelector, // should be same as selector added on controller
 			},
 		},
-		Spec: aadpodv1.AzureIdentityBindingSpec{
-			AzureIdentity: copiedIdentity.Name,
-			Selector:      infrav1.AzureIdentityBindingSelector, // should be same as selector added on controller
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "AzureIdentityBinding",
```

Comment:
I tested again whether the NMI pod intercepts any requests that it doesn't need to and I only saw it intercept requests from ASO when UserAssignedMSI auth is in use, which is what I hoped to see.